src/Tests/Microsoft.Diagnostics.Monitoring.StartupHook.UnitTests/Exceptions/Pipeline/Steps/ExceptionDemultiplexerPipelineStepTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using System.Collections.Generic;
+using System;
+using Xunit;
+
+namespace Microsoft.Diagnostics.Monitoring.StartupHook.Exceptions.Pipeline.Steps
+{
+    [TargetFrameworkMonikerTrait(TargetFrameworkMonikerExtensions.CurrentTargetFrameworkMoniker)]
+    public sealed class ExceptionDemultiplexerPipelineStepTests
+    {
+        private readonly Dictionary<Exception, int> _reportedExceptionCounts = new();
```

Comment:
Right now, it looks like we only ever have a single instance of each exception, which makes this feels overengineered - I'm assuming in the future this mechanism will be leveraged more

Reply:
I was potentially thinking of adding more tests that would flex this a bit more, but decided against it for now. The way it is allows us to determine the exact count of each type of exception that is reported rather than failing the step's invocation immediately.